shared/folders/files/index.js

```diff
@@ -1,14 +1,94 @@
 // @flow
 import React, {Component} from 'react'
 import {connect} from 'react-redux'
-import {Text} from '../../common-adapters'
+import {bindActionCreators} from 'redux'
+import {openInKBFS} from '../../actions/kbfs'
+import {navigateBack} from '../../actions/router'
+import flags from '../../util/feature-flags'
+import Render from './render'
+import _ from 'lodash'
 
-// import Render from './render'
+import type {Folder} from '../list'
+
+type Props = $Shape<{
+  folder: ?Folder,
+  path: string,
+  username: string,
+  navigateBack: () => void,
+  openInKBFS: (path: string) => void
+}>
+
+type State = {
+  visiblePopupMenu: boolean
+};
+
+class Files extends Component<void, Props, State> {
+  state: State;
+
+  _checkFolderExistance (props) {
+    // TODO (AW): make a more user friendly response for when the folder they were hoping to look at has been removed
+    if (!props.folder) props.navigateBack()
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      visiblePopupMenu: false
+    }
+    this._checkFolderExistance(props)
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this._checkFolderExistance(nextProps)
+  }
 
-class Files extends Component {
   render () {
-    return <Text type='Body'>TODO</Text>
+    const {folder, username} = this.props
+    if (!folder) return null // Protect from state where the folder to be displayed was removed
+    const openCurrentFolder = () => { this.props.openInKBFS(this.props.path) }
+    const ignoreCurrentFolder = () => { console.log('TODO (AW): implement ignore folder') } // TODO (AW): implement ignore file
+    return (
+      <Render
+        theme={folder.isPublic ? 'public' : 'private'}
+        popupMenuItems={[
+          {onClick: openCurrentFolder, title: 'Open folder'},
+          {onClick: ignoreCurrentFolder, title: 'Ignore'}
+        ]}
+        visiblePopupMenu={this.state.visiblePopupMenu}
+        onTogglePopupMenu={() => this.setState({visiblePopupMenu: !this.state.visiblePopupMenu})}
+        selfUsername={username}
+        users={folder.users}
+        waitingForParticipantUnlock={[]}
```

Comment:
I think if we're going to integrate the smart component we should integrate the whole thing. if there's parts that aren't implemented we should do that in the correct place and not here. so i'd actually hook this up to the store and have the reducer just init it to [], but have that plumbed through as a prop here. otherwise we'll have this half-integrated component. same for the ignoreCurrentFolder above, we should make the action creator and put the TODO there